### PR TITLE
MAINT: Cleaning up PY_MAJOR_VERSION/PY_VERSION_HEX

### PR DESCRIPTION
**Pull request by sethtroisi**
Remove most of PY_MAJOR_VERSION guards that are no longer needed (after Python2 support is dropped)

There are a handful more of these that are more difficult (e.g. [this](https://github.com/numpy/numpy/blob/f91b033aa35b929610c0db12f16b1b0c1ddc08e6/numpy/core/src/multiarray/typeinfo.c#L108) or [this](https://github.com/numpy/numpy/blob/96708210309be41cb7bba85155416fc0fc3f86a3/numpy/core/src/multiarray/common.c#L169)) that will be handled in a 2nd pass

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
Looks good on a first pass, will let a second reviewer confirm

File: numpy/core/include/numpy/npy_3kcompat.h [hunk 1]
```diff
@@ -13,11 +13,9 @@
 #include <Python.h>
 #include <stdio.h>
 
-#if PY_VERSION_HEX >= 0x03000000
 #ifndef NPY_PY3K
 #define NPY_PY3K 1
```
Comment (seberg) [inline numpy/core/include/numpy/npy_3kcompat.h]: I removing the internal use of this is one of the next steps.

**Review by seberg — COMMENTED** [review]
All changes here look good to me as well. Note that we have `#if PY_VERSION_HEX < 0x03030000` and `#if PY_VERSION_HEX >= 0x03050000` which also apply to 1.16+ (and 1.17). So that we can remove those just as well. (and one `PY_VERSION_HEX < 0x03040400`).

I am fine with moving that to a new PR though if you prefer.

File: numpy/core/include/numpy/npy_3kcompat.h (hunk 1 above)
Comment (sethtroisi) [reply numpy/core/include/numpy/npy_3kcompat.h]: Yes, I'm also working on cleaning that up (in another CL)

**Review by sethtroisi — COMMENTED** [review]
> All changes here look good to me as well. Note that we have `#if PY_VERSION_HEX < 0x03030000` and `#if PY_VERSION_HEX >= 0x03050000` which also apply to 1.16+ (and 1.17). So that we can remove those just as well. (and one `PY_VERSION_HEX < 0x03040400`).
> 
> I am fine with moving that to a new PR though if you prefer.

I'm doing those in a separate CL as you have to inspect and process the logic while this is just verify matching #if #endif

**Comment by sethtroisi** [comment]
I think this can be pulled. (I don't want to sound pushy, I'm just excited to build the other PRs off this)

**Comment by mattip** [comment]
One of these PRs should add a release note

**Comment by mattip** [comment]
Thanks @sethtroisi

**Pull request merged**